satpy/modifiers/_crefl_utils.py

```diff
@@ -49,6 +88,443 @@
 REFLMAX = 1.6
 
 
+class _Coefficients:
+    LUTS: list[np.ndarray] = []
+    # resolution -> wavelength -> coefficient index
+    # resolution -> band name -> coefficient index
+    COEFF_INDEX_MAP: dict[int, dict[Union[tuple, str], int]] = {}
+
+    def __init__(self, wavelength_range, resolution=0):
+        self._wv_range = wavelength_range
+        self._resolution = resolution
+
+    def __call__(self):
+        idx = self._find_coefficient_index(self._wv_range, resolution=self._resolution)
+        band_luts = [lut_array[idx] for lut_array in self.LUTS]
+        return band_luts
+
+    def _find_coefficient_index(self, wavelength_range, resolution=0):
+        """Return index in to coefficient arrays for this band's wavelength.
+
+        This function search through the `COEFF_INDEX_MAP` dictionary and
+        finds the first key where the nominal wavelength of `wavelength_range`
+        falls between the minimum wavelength and maximum wavelength of the key.
+        `wavelength_range` can also be the standard name of the band. For
+        example, "M05" for VIIRS or "1" for MODIS.
+
+        :param wavelength_range: 3-element tuple of (min wavelength, nominal wavelength, max wavelength)
+        :param resolution: resolution of the band to be corrected
+        :return: index in to coefficient arrays like `aH2O`, `aO3`, etc.
+                 None is returned if no matching wavelength is found
+        """
+        index_map = self.COEFF_INDEX_MAP
+        # Find the best resolution of coefficients
+        for res in sorted(index_map.keys()):
+            if resolution <= res:
+                index_map = index_map[res]
+                break
+        else:
+            raise ValueError("Unrecognized data resolution: {}", resolution)
+        # Find the best wavelength of coefficients
+        if isinstance(wavelength_range, str):
+            # wavelength range is actually a band name
+            return index_map[wavelength_range]
+        for k, v in index_map.items():
+            if isinstance(k, str):
+                # we are analyzing wavelengths and ignoring dataset names
+                continue
+            if k[0] <= wavelength_range[1] <= k[2]:
```

Comment:
We have a class that does this... https://github.com/pytroll/satpy/blob/main/satpy/dataset/dataid.py#L100

Reply:
WavelengthRange is now being used.